lib/index.js

```diff
@@ -79,20 +79,30 @@ PostHTML.prototype.process = function (tree, options) {
    * @prop {?Boolean} options.sync - enables sync mode, plugins will run synchronously, throws an error when used with async plugins
    * @prop {?Function} options.parser - use custom parser, replaces default (posthtml-parser)
    * @prop {?Function} options.render - use custom render, replaces default (posthtml-render)
-   * @prop {?Boolean} options.skipParse - disable parsing
    */
   options = options || {}
 
+  if (typeof tree !== 'string' || typeof tree !== 'object') {
+    throw new TypeError(
+      'Input must to be a String (HTML) or Object (PostHTML AST)'
+    )
+  }
+
+  tree = typeof tree === 'object' ? tree : parser(tree)
+
+  tree.parser = parser
```

Comment:
i create a new issue about this changes for parser/render options

```js
var postHTML = function(plugins) { this.plugins = plugins; };

postHTML.prototype.parser = function(html) { return 'OLD' + html; }; // CHANGE
postHTML.prototype.process = function(html, options) {
  var tree = this.parser(html);
  if (!this.plugins.length)
    return tree;
  
  this.plugins.forEach(function(plug) {
    tree = plug(tree);
  });
  
  return tree;
  
}

var ph = function(plugins) {
  return new postHTML(plugins);
}

var pDefault = ph([function(s) { return s + 'DEFAULT_PLUGINS'; }]);
var pCustom = ph([function(s) { return s + 'CUSTOM_PLUGINS'; }]);


pCustom.parser = function customParser(html) {return html + 'NEW'; };

console.log(pDefault.parser('sss'));
console.log(pDefault.process('sss'));

console.log(pCustom.parser('sss'));
console.log(pCustom.process('sss'));

console.log('\n\n\nchain', pDefault.process(pCustom.process('sss')));
```

in this issue should be stay changet about messages API. Ok?